### DISCUSSION TITLE
Limit RGB LED currents separately from white CCT LEDs?

### DISCUSSION THREAD

**Discussion by ManfredBartz** [arendst/Tasmota#17866] [Q&A] [answered]
I have configured an **ESP32-C3** based module for a **SM2135** based CCTRGB downlight from "ARLEC".

To match the currents measured with the original light, I need different PWM values.
For CCT it should be 24mA but I get 21mA with _SM2135 DAT3_ and 30mA with _SM2135 DAT4_.
I suppose I could use **SM2135 DAT4** and limit the dimmer range?

However, for **R** it should be 15mA, for **G** 11mA and for **B** 12mA but I get 21mA at dimmer=100% with _SM2135 DAT3_.
Is there a way to limit RGB separately from CCT?
Do I need to learn Berry scripting?
Any hints?

**Comment by barbudor** [comment, marked as answer]
Depending on the index chosen you get a different combination:
| Index | RGB | CCT |
|--|--|--|
| 1 or 2 | 20mA | 15mA |
| 3 or 4 | 20mA | 30mA |
| 5 or 6 | 45mA | 60mA |
| 7  | 15mA | 40mA |
| 8 | reserved |
| 9 or 10 | 20mA | 15mA |

You are not supposed to have different current between DAT3 and DAT4, only a different led order (even: BGR, odd:GRB)

1) You can't set separately R, G, B => that's a chip limitation
2) You are limited to the above combinations => that's a tasmota limitation

With Berry you could re-write a driver to overcome limitation 2)
But I fear it won't be integrated in Tasmota Light model and you would have to re-code all commands (dimmer, color...)

It should be possible to extend the current model past Index 10 to add an new combination

And of course as you pointed, you can also use the upper level and limit the range
